src/dmd/expressionsem.d

```diff
@@ -6382,6 +6382,36 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
         if (exp.arrow) // ImportC only
             exp.e1 = exp.e1.expressionSemantic(sc).arrayFuncConv(sc);
+
+        if (sc.flags & SCOPE.Cfile)
+        {
+            if (exp.ident == Id.__xalignof && exp.e1.isTypeExp())
+            {
+                // C11 6.5.3 says _Alignof only applies to types
+                Expression e;
+                Type t;
+                Dsymbol s;
+                dmd.typesem.resolve(exp.e1.type, exp.e1.loc, sc, e, t, s, true);
+                if (e)
+                {
+                    exp.e1.error("argument to `_Alignof` must be a type");
+                    return setError();
+                }
+                else if (t)
+                {
+                    result = new IntegerExp(exp.loc, t.alignsize, Type.tsize_t);
+                }
+                else if (s)
+                {
+                    exp.e1.error("argument to `_Alignof` must be a type");
+                    return setError();
+                }
+                else
+                    assert(0);
```

Comment:
This entire if-else structure can be rewritten to:

```d
if (e || s)
{
    exp.e1.error("argument to `_Alignof` must be a type");
    return setError();
}
else if (t)
{
     result = new IntegerExp(exp.loc, t.alignsize, Type.tsize_t);
}
else
{
    assert(0);
}
```

Reply:
the order is important, which is why I wrote it like that.

Reply:
Then maybe a test that depends on this order should be added. Right now, the `else if (s)` branch is not covered.

Reply:
All the callers of resolve() follow the same order. I'm actually not sure it is possible to trip the (s) branch, as C doesn't really have symbols that aren't expressions or types.

Reply:
> All the callers of resolve() follow the same order. I'm actually not sure it is possible to trip the (s) branch, as C doesn't really have symbols that aren't expressions or types.

`_Alignof(test) `?

Reply:
`test` is a function - an expression